test/heap-types.wast

```diff
@@ -167,5 +167,16 @@
     (drop
       (ref.cast $struct.B (ref.null $struct.A) (rtt.canon $struct.B))
     )
+    (drop
+      (block $out (result (ref $struct.B))
+        (drop
+          (br_on_cast $out $struct.B (ref.null $struct.A) (rtt.canon $struct.B))
+        )
+        ;; an untaken br_on_cast, with unreachable rtt - so we cannot use the
+        ;; RTT in binaryen IR to find the cast type.
+        (br_on_cast $out $struct.B (ref.null $struct.A) (unreachable))
+        (unreachable)
+      )
+    )
```

Comment:
I'd be good to add the test that `block`'s return type that's targeted by `br_on_cast` is correct, as in https://github.com/WebAssembly/binaryen/pull/3451#issuecomment-747171175.

Reply:
Sorry, I must be missing the point of that comment... I thought you were saying to check that we send the right type to the block. Isn't that what's tested on https://github.com/WebAssembly/binaryen/pull/3451/files#diff-02d11b1ea6faf71dddbf6ba74127bc9022481aeb27d74e423821c029cf814ad2R156 ?

Reply:
That value is dropped, so even if it is not the right type, it's not gonna be caught.. no?

Reply:
Is your concern that the value that falls through doesn't have the proper type, which in that case should be  `anyref`? And that something other than a drop could notice that the type is not `anyref`?

Reply:
Yeah as in my example, can we pass the block as a function argument that expects just `ref $struct.B` and not `anyref` or `ref $struct.A`? (I think we can, but anyway, what I meant was a test for that)

Reply:
I'm confused then, maybe because it's late...

Your example is
```wat
(func $foo (param (ref $struct.B))

(call $foo
  (block $out (result (ref $struct.B))
    (drop
      (br_on_cast $out $struct.B (ref.null $struct.A) (rtt.canon $struct.B))
    )
    (unreachable)
  )
)
```
I'm not sure what role the `call` plays here? The block having result `$struct.B` already is enough by itself to verify that we send that type to the block. That is, all branches to `$out` must have type `$struct.B` (or an even more refined type, but there isn't one in the program) or it won't validate? So I think that was tested already, unless I'm missing something.

Reply:
Oh, right, the validator checks that... I forgot about that sorry. Sorry for making a lot of noise at night. 😅

Reply:
No problem!

And good we looked at this more carefully as the other test was indeed missing for the value flowing out (as the drop accepted any type).